### PR TITLE
Number Pairs

### PR DESCRIPTION
**Pull request by VolhaVaitulevich**
Task description: https://edabit.com/challenge/fpHGmHoKX5RyD3wjQ

### CODE REVIEW

File: exercises/arrays/numberPairs/index.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const numberPairs = (str: string): number => {
+	const arrOfNumbers = str.split(' ').slice(1);
+	const objOfDuplicates = arrOfNumbers.reduce(
+		(acc: Record<string, number>, item): Record<string, number> => {
+			acc[item] = +arrOfNumbers.filter(num => num === item).length;
+			return acc;
+		},
+		{},
+	);
+	console.log(objOfDuplicates);
```
Comment (siarhei-papkou) [inline exercises/arrays/numberPairs/index.ts]: please remove

File: exercises/arrays/numberPairs/index.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const numberPairs = (str: string): number => {
+	const arrOfNumbers = str.split(' ').slice(1);
+	const objOfDuplicates = arrOfNumbers.reduce(
+		(acc: Record<string, number>, item): Record<string, number> => {
+			acc[item] = +arrOfNumbers.filter(num => num === item).length;
```
Comment (siarhei-papkou) [inline exercises/arrays/numberPairs/index.ts]: Let's avoid nested loops. I guess we can find a better solution ;)

File: exercises/arrays/numberPairs/index.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+			return acc;
+		},
+		{},
+	);
+	console.log(objOfDuplicates);
+	let numberOfPairs = 0;
+	for (const item in objOfDuplicates) {
```
Comment (siarhei-papkou) [inline exercises/arrays/numberPairs/index.ts]: Try to use array method instead of `for`

**Review by siarhei-papkou — CHANGES_REQUESTED** [review]

File: exercises/arrays/numberPairs/index.ts (hunk 3 above)
Comment (VolhaVaitulevich) [reply exercises/arrays/numberPairs/index.ts]: Changed it to 'reduce', made a new commit

File: exercises/arrays/numberPairs/index.ts (hunk 2 above)
Comment (VolhaVaitulevich) [reply exercises/arrays/numberPairs/index.ts]: Removed `filter`, added `hasOwnProperty`, and made a new commit.

**Review by siarhei-papkou — APPROVED** [review]

**Pull request merged**